boards/ti/lp_em_cc2340r5/lp_em_cc2340r5_defconfig

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2024 Texas Instruments Incorporated
+# Copyright (c) 2024 BayLibre, SAS
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Ensure hex-file is not filled between lower and upper flash.
+# If not, it will grow to 3.5 GB.
+CONFIG_BUILD_NO_GAP_FILL=y
+CONFIG_BUILD_OUTPUT_HEX=y
+
+CONFIG_GPIO=y
+CONFIG_PINCTRL=y
```

Comment:
pinctrl not allowed in boards

Reply:
Are you sure ?
I ask because there many occurrences: 319 in boards directory, among which 7 Nordic boards. :)

Any other way to get it enabled by default ?

Reply:
Draft PR updated. I took into account your comments @nordicjm , except this one about CONFIG_PINCTRL (I did not ignore/forget it, it's just because the discussion is still open).

Reply:
It needs to be removed, drivers need to select it in their Kconfig if it is required

Reply:
Understood, that makes sense. Thank you for this suggestion.
We will fix it at next PR update, while fixing the others issues reported by CI.